File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  location_grouped: string;
  Furnishing: string;
  Transaction: string;
  Ownership: string;
  facing: string;
  carpet_area_sqft: number;
  floor_num: number;
  bathroom: number;
  balcony: number;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class PredictionRequest(BaseModel):
    location_grouped: str
    Furnishing: str
    Transaction: str
    Ownership: str
    facing: str
    carpet_area_sqft: float
    floor_num: float
    bathroom: float
    balcony: float


class PredictionResponse(BaseModel):
    predicted_price: float
    